src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6828,11 +6907,13 @@ public override void VisitForEachIterationVariables(BoundForEachStatement node)
                         SetAnalyzedNullability(node.IterationVariableType, new VisitResult(result, destinationType), isLvalue: true);
                         state = result.State;
 
-                        // If we inferred the type of this variable, then record the inferred type of the variable for later use by the SemanticModel.
+                        // Update the variable type and the containing symbol for use in the semantic model.
+                        var variableType = iterationVariable.TypeWithAnnotations;
                         if (node.Syntax is ForEachStatementSyntax { Type: { IsVar: true } })
                         {
-                            _variableTypes[iterationVariable] = result.ToTypeWithAnnotations();
+                            variableType = result.ToTypeWithAnnotations();
                         }
+                        SetLocalType(node, iterationVariable, variableType);
```

Comment:
>SetLocalType(node, iterationVariable, variableType); [](start = 24, length = 52)

Is there a guarantee that we are dealing with "var" variable?

Reply:
No longer relevant.

---
In reply to: [327924666](https://github.com/dotnet/roslyn/pull/38769#discussion_r327924666) [](ancestors = 327924666)